socialhome/streams/app/stores/streamStore.js

```diff
@@ -13,7 +13,23 @@ Vue.use(Vuex)
 function onSuccess(state, payload) {
```

Comment:
Actually, thinking about this, as I was starting to split it up. Shouldn't we not be using an `onSuccess` at all? AFAICT we should be putting all this code into our mutations. What do you think? I know this was a first version, just thinking if it's correct to keep as we move forward.

Reply:
OK, ignore that last comment. Did some playing around and I think the solution you've provided before is the most optimum one.

Also, the feedback about `contents` not having everything was good. It is cleaner to have them separate, so I did some refactoring. To avoid code duplication I had to go a bit deeper, touching quite a few components in the process. Only broke 27 tests :D Will fix tomorrow and hopefully get this finally merged. Re-review welcome.